contracts/exchange-forwarder/contracts/src/MixinExchangeWrapper.sol

```diff
@@ -494,4 +496,27 @@ contract MixinExchangeWrapper is
     {
         return order.makerFeeAssetData.length > 3 && order.makerFeeAssetData.readBytes4(0) == EXCHANGE_V2_ORDER_ID;
     }
+
+    /// @dev Checks whether one asset is effectively equal to another asset.
+    ///      This is the case if they have the same ERC20Proxy/ERC20BridgeProxy asset data, or if
+    ///      one is the ERC20Bridge equivalent of the other.
+    /// @param takerFee Byte array encoded for the takerFee asset proxy.
+    /// @param takerFeeAssetData Byte array encoded for the maker asset proxy.
+    /// @return Whether or not the underlying assets are equal.
+    function _noTakerFee(
```

Comment:
What do we actually gain from special casing the `StaticCallProxy` here? I feel like it makes the logic inconsistent from how the `StaticCallProxy` works elsewhere. Also in the interest of keeping this codebase as lean as possible, I'd probably opt to remove all of these checks completely.

Reply:
If the `takerFeeAssetData` is StaticCall and we _don't_ treat the `takerFeePaid` amount as zero, we'd be subtracting some arbitrary non-zero amount from either the `wethSpentAmount` or the `makerAssetAcquiredAmount`; either way the results would be incorrect/misleading. So the options are either:
- this (`_noTakerFee`)
- adding another `else if` block to `_marketSellSingleOrder` and `_marketBuySingleOrder`
- enforcing that the `takerFeeAssetData` is not StaticCall

I think the current option is a bit cleaner than the 2nd option, but I'd be down for the 3rd.

Reply:
The different scenarios where this is currently used are:

- marketBuy where `order.takerFee == 0`. `takerFeePaid` will always be 0 here anyways.
- marketBuy where both `order.takerFeeAssetData` and `order.makerAssetData` are the same `StaticCall` data. The amount doesn't really matter in the case, so I don't see a huge need for extra logic.
- marketSell where `order.takerFee == 0`. `takerFeePaid` will always be 0 here anyways.
- marketSell where both `order.takerFeeAssetData` and `order.takerAssetData` are the same `StaticCall` data. Note this case will never be hit in the current contract because `takerAssetData` is always WETH. In future iterations where this is possible, the amount also doesn't really matter 🤷‍♂ .

I know it's kind of funky, but I think it's simpler to be as agnostic as possible when working with different `assetData` formats and only special case things when absolutely necessary. Otherwise the code becomes much less maintainable, so I'd still argue for removing this.